packages/files/file.pony

```diff
@@ -212,19 +203,19 @@ class File
       return FileError
     end
 
-    
+
   fun valid(): Bool =>
     """
     Returns true if the file is currently open.
     """
-    not _handle.is_null()
+    not _fd == -1
 
   fun get_fd(): I32 ? =>
     """
     Returns the underlying file descriptor.
     Raises an error if the file is not currently open.
     """
-    if _handle.is_null() then
+    if _fd == -1 then
```

Comment:
Making the change is not an issue. I had tried to keep all the functions exactly the same semantically to not change the API at all since that would trigger an `rfc` process.